File: Examples/Ruby/Links/get_page_link_annotations.rb
require 'aspose_pdf_cloud'

class Links

  include AsposePDFCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @pdf_api = PdfApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def get_page_link_annotations
    file_name = "Sample-Bookmark.pdf"
    upload_file(file_name)

    response = @pdf_api.get_document_bookmarks(file_name)
  end

end

links = Links.new()
puts links.get_page_link_annotations

File: Examples/PHP/Links/GetPageLinkAnnotations.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\PDF\PdfApi;
use Aspose\PDF\AsposeApp;

class Links {

	public $pdfApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->pdfApi = new PdfApi();
	}

	public function getPageLinkAnnotations() {
		// Upload file to Aspose Cloud Storage
		$fileName = "Sample-Bookmark.pdf";
		Utils::uploadFile($fileName);
		
		$result = $this->pdfApi->GetPageLinkAnnotations($fileName, $pageNumber = 1, $storage = "", $folder = "");
		print_r ( $result );
	}
}

$links = new Links();
$links->getPageLinkAnnotations();

?>
//ExEnd:
